### ISSUE TITLE
Bug: Table Attribute Potential Bug

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#464]
This issue needs a validation from the community - if this is really a big case or not. The priority will vary on the community request.

The `Table` attribute of `System.ComponentModel.DataAnnotations.Schema` is expecting the `Table("TableName")`, but there is a possibility the the `Schema` property may had been set explicitly. In RepoDb, this attribute is seems to be used as `Table("[Schema].[TableName]")` and defaultly ignoring the `Schema` property, like the existing `Map("[Schema].[TableName]")`.

Here is the [link](https://github.com/mikependon/RepoDb/blob/4719afa9054c1dc04de30c66fc7648ee2b10f0ea/RepoDb.Core/RepoDb/Extensions/DataEntityExtension.cs#L49) to the code that could have the potential bug.

I would suggest to write it this way.

```csharp
internal static string GetMappedName(Type type)
{
	return type.GetCustomAttribute<MapAttribute>()?.Name ??
		   GetMappedName(type.GetCustomAttribute<TableAttribute>()) ??
		   ClassMapper.Get(type) ??
		   type.Name;
}
```

With a helper method like below.

```csharp
private static string GetMappedName(Table tableAttribute)
{
	if (tableAttribute == null)
	{
		return null;
	}
	if (string.IsNullOrEmpty(tableAttribute.Schema))
	{
		return tableAttribute.Name;
	}
	return $"[{tableAttribute.Schema}].[{tableAttribute.Name}]";
}
```

**Important Note:** We need to bring the `IDbSetting` object and use the `ClosingQuote` and `OpeningQuote` property to quote the values.

**Acceptance Criteria:** Make sure that all the existing Unit Tests and Integration Tests of all Solutions are passing (i.e.: RepoDb.Core, RepoDb.SqlLite, RepoDb.SqlServer, RepoDb.PostgreSql, RepoDb.MySql, RepoDb.MySqlConnector and RepoDb.SqlServer.BulkOperations).

**Comment by mikependon** [comment]
The fix is now available at [RepoDb v1.11.4-beta3](https://www.nuget.org/packages/RepoDb/1.11.4-beta3).

**Comment by mikependon** [comment]
Closing this issue now.
